platform/ios/src/MGLMapView.mm

```diff
@@ -4249,11 +4249,7 @@ - (void)setSelectedAnnotations:(NS_ARRAY_OF(id <MGLAnnotation>) *)selectedAnnota
 
     if ([firstAnnotation isKindOfClass:[MGLMultiPoint class]]) return;
 
-    // Select the annotation if it’s visible.
-    if (MGLCoordinateInCoordinateBounds(firstAnnotation.coordinate, self.visibleCoordinateBounds))
-    {
-        [self selectAnnotation:firstAnnotation animated:NO];
-    }
+    [self selectAnnotation:firstAnnotation animated:NO];
```

Comment:
Also, this documentation needs to be updated:

https://github.com/mapbox/mapbox-gl-native/blob/9fa27497059f9c08144e49a411bd735efb680fb4/platform/ios/src/MGLMapView.h#L1171-L1172